client/matching/client.go

```diff
@@ -152,6 +152,8 @@ func (c *clientImpl) PollForActivityTask(
 		partition,
 		resp.LoadBalancerHints,
 	)
+	// caller needs to know the actual partition for cancelling long poll, so modify the request to pass this information
```

Comment:
I don't like this pattern. I think we should return the partition explicitly to the caller, but for a quick fix, I'll go with this way.